internal/controller/registrycache/registry_cache_config_controller.go

```diff
@@ -68,17 +69,28 @@ func (r *RegistryCacheConfigReconciler) Reconcile(ctx context.Context, request c
 		return requeueOnError(err)
 	}
 
-	var runtime imv1.Runtime
-	if err := r.KcpClient.Get(ctx, types.NamespacedName{Name: runtimeID,
-		Namespace: request.Namespace,
-	}, &runtime); err != nil {
+	var runtimeList imv1.RuntimeList
+
+	err = r.KcpClient.List(ctx, &runtimeList,
+		client.MatchingLabels(map[string]string{RuntimeIDLabel: runtimeID}))
+
+	if err != nil {
+		r.Log.V(log_level.TRACE).Error(err, "Failed to find runtime", "RuntimeID", runtimeID, "Namespace", secret.Namespace)
 		return requeueOnError(err)
 	}
 
-	if registryCacheEnabled || len(runtime.Spec.Caching) > 0 {
+	if len(runtimeList.Items) == 0 || len(runtimeList.Items) > 1 {
+		e := fmt.Errorf("expected to find one runtime for given runtime ID, found %d", len(runtimeList.Items))
+		r.Log.V(log_level.TRACE).Error(e, "RuntimeID", runtimeID, "Namespace", secret.Namespace)
```

Comment:
The errors in this package could be muted because they are logged at `log_level.TRACE`, just to double-check if this is intentional